src/libs/actions/IOU.js

```diff
@@ -720,11 +720,192 @@ function payIOUReport({
     return promiseWithHandlers;
 }
 
+/**
+ * @param {String} chatReportID - Report ID of the chat where the IOU is.
+ * @param {Number} amount - IOU amount in cents.
+ * @param {String} currency - IOU currency.
+ * @param {String} comment
+ * @param {Array} participants - Participants on the IOU
+ */
+function sendMoneyWithWallet(chatReportID, amount, currency, comment, participants) {
+    const chatReport = _.empty(chatReportID) ? chatReports[`${ONYXKEYS.COLLECTION.REPORT}${chatReportID}`] : ReportUtils.buildOptimisticChatReport(participants);
+
+    const ownerEmail = participants[0].login;
+    const userEmail = participants[1].login;
+    const optimisticIOUReport = ReportUtils.buildOptimisticIOUReport(
+        ownerEmail,
+        userEmail,
+        amount,
+        chatReport.chatReportID,
+        currency,
+        preferredLocale,
+    );
+
+    const newSequenceNumber = Report.getMaxSequenceNumber(chatReport.reportID) + 1;
+    const optimisticIOUReportAction = ReportUtils.buildOptimisticIOUReportAction(
+        newSequenceNumber,
+        CONST.IOU.REPORT_ACTION_TYPE.PAY,
+        amount,
+        currency,
+        comment,
+        participants,
+        optimisticIOUReport.reportID,
+    );
```

Comment:
Do you mean this one @mountiny ?

Reply:
@sketchydroide  Yes, just this one will be needed. We might need to update the function tho so it created the Sending Pay action correctly, I dont think it does now (it did not have to til now)